app/src/lib/git/status.ts

```diff
@@ -198,3 +163,91 @@ export async function getStatus(
     workingDirectory,
   }
 }
+
+function addEntryToFiles(
+  files: Map<string, WorkingDirectoryFileChange>,
+  entry: IStatusEntry,
+  filesWithConflictMarkers: Set<string>
+) {
+  const status = mapStatus(entry.statusCode)
+  const hasConflictMarkers = filesWithConflictMarkers.has(entry.path)
+  if (status.kind === 'ordinary') {
+    // when a file is added in the index but then removed in the working
+    // directory, the file won't be part of the commit, so we can skip
+    // displaying this entry in the changes list
+    if (
+      status.index === GitStatusEntry.Added &&
+      status.workingTree === GitStatusEntry.Deleted
+    ) {
+      return files
+    }
+  }
+
+  if (status.kind === 'untracked') {
+    // when a delete has been staged, but an untracked file exists with the
+    // same path, we should ensure that we only draw one entry in the
+    // changes list - see if an entry already exists for this path and
+    // remove it if found
+    files.delete(entry.path)
+  }
+
+  // for now we just poke at the existing summary
+  const summary = convertToAppStatus(status, hasConflictMarkers)
+  const selection = DiffSelection.fromInitialSelection(DiffSelectionType.All)
+
+  files.set(
+    entry.path,
+    new WorkingDirectoryFileChange(
+      entry.path,
+      summary,
+      selection,
+      entry.oldPath
+    )
+  )
+  return files
+}
+
+function handleHeader(
```

Comment:
:art: similarly, documentation for this function? :memo:

Reply:
18251e8